[DEVX-2107] Add testcafe docs about ts compiler options (#1695)
saucelabs/sauce-docs: docs/web-apps/automated-testing/testcafe/yaml.md

<p><small>| OPTIONAL | STRING |</small></p>

> This field has been deprecated as of TestCafe v1.10.0. See [TestCafe Documentation](https://testcafe.io/documentation/402638/reference/configuration-file#tsconfigpath). Please refer to [compilerOptions](#compilerOptions).

Reviewer: lowercase required for on page links.
```suggestion
This field has been deprecated as of TestCafe v1.10.0. See [TestCafe Documentation](https://testcafe.io/documentation/402638/reference/configuration-file#tsconfigpath). Please refer to [compilerOptions](#compileroptions).
```

> This field has been deprecated as of TestCafe v.1.10.0. See [TestCafe Documentation](https://testcafe.io/documentation/402638/reference/configuration-file#tsconfigpath). Please refer to [compilerOptions](#compilerOptions).

Reviewer 2: ```suggestion
This field has been deprecated as of TestCafe v1.10.0. See [TestCafe Documentation](https://testcafe.io/documentation/402638/reference/configuration-file#tsconfigpath). Please refer to [compilerOptions](#compilerOptions).
```

---

### `compilerOptions`

<p><small>| OPTIONAL | OBJECT |</small></p>

Specifies test compilation settings. The current version of TestCafe can only configure the TypeScript compiler. See [TestCafe Documentation](https://testcafe.io/documentation/402638/reference/configuration-file#compileroptions).

```yaml
  compilerOptions:
    configPath: /path/to/tsconfig.json
    customCompilerModulePath: ../typescript@4
```
---

#### `configPath`

<p><small>| OPTIONAL | STRING |</small></p>

The absolute or relative path to the TypeScript configuration file. Relative paths are resolved against the current directory (the directory from which you run TestCafe).

```yaml
    configPath: /path/to/tsconfig.json
```

:::note

> We recommend that you avoid the use of special characters when naming your config file. It may cause issue to launch TestCafe test.

Reviewer 2: ```suggestion
We recommend that you avoid the use of special characters when naming your config file. It may prevent TestCafe tests from launching.
```

:::

---

#### `customCompilerModulePath`

<p><small>| OPTIONAL | STRING |</small></p>

TestCafe ships with a `typescript@3` compiler. This field is for compiling your tests with a different compiler.

```yaml
    customCompilerModulePath: ../typescript@4
```

---

#### `options`

<p><small>| OPTIONAL | OBJECT |</small></p>

> Specifies the compiler options listed in the official [TypeScript documentation](https://www.typescriptlang.org/docs/handbook/compiler-options.html).

Reviewer: ```suggestion
Specifies the compiler options. See [TypeScript documentation](https://www.typescriptlang.org/docs/handbook/compiler-options.html).
```

```yaml
    options:
      showConfig: true
```

:::note

> We recommend that you avoid the use of special characters when naming your config file. It may cause issue to launch TestCafe test.

Reviewer 2: ```suggestion
We recommend that you avoid the use of special characters when naming your config file. It may prevent TestCafe tests from launching.
```
